### PR TITLE
WIP: feat(fix): add option to fix errors on save

### PR DESCRIPTION
**Pull request by Maistho**
Adds an option to fix errors on saving.

The option defaults to false.

### CODE REVIEW

**Comment by Maistho** [comment]
Just noticed that it runs on non-TS-files currently. I'll add a test case and fix it, but don't merge it yet.

File: lib/main.js [hunk 1]
```diff
@@ -63,6 +63,18 @@ export default {
         this.useLocalTslint = use;
       }),
     );
+
+    this.subscriptions.add(
+      atom.workspace.observeTextEditors((editor) => {
+        editor.onDidSave(() => {
+          if (atom.config.get('linter-tslint.fixOnSave')) {
```
Comment (Xapphire13) [inline lib/main.js]: Your issue is that you need to check the grammar scope on the saving editor, not only `fixOnSave`

File: lib/main.js [hunk 2]
```diff
@@ -131,6 +131,21 @@ export default {
...
       }),
     );
+
+    this.subscriptions.add(
+      atom.workspace.observeTextEditors((editor) => {
+        editor.onDidSave(() => {
+          const validScope = editor.getCursors().some(cursor =>
+            cursor.getScopeDescriptor().getScopesArray().some(scope =>
+              grammarScopes.includes(scope)));
+          if (validScope && atom.config.get('linter-tslint.fixOnSave')) {
```
Comment (Maistho) [inline lib/main.js]: @Xapphire13 

This ought to solve that right?

Comment (Xapphire13) [inline lib/main.js]: That should work, but for consistency with the code I recently added could you make sure it works with `textEditor.getRootScopeDescriptor().getScopesArray()`?

Comment (Xapphire13) [inline lib/main.js]: Actually, we can avoid having another `textEditors` subscription by adding this logic [here](https://github.com/AtomLinter/linter-tslint/blob/master/lib/main.js#L94)

Comment (Maistho) [inline lib/main.js]: Oh, that makes a lot more sense. Change incoming

**Comment by Maistho** [comment]
Thanks for the help during the PR! 👍

**Comment by Xapphire13** [comment]
Anytime =]

**Pull request merged**